server/src/lib/model/db.ts

```diff
@@ -678,4 +679,28 @@ export default class DB {
     );
     return row;
   }
+
+  async hasChallengeEnded(challenge: ChallengeToken) {
+    let hasChallengeEnded = true;
+    const [[row]] = await this.mysql.query(
+      `SELECT TIMESTAMPADD(MINUTE, -TIMESTAMPDIFF(MINUTE, UTC_TIMESTAMP(), NOW()), start_date) AS start_date_utc
+      FROM challenges
+      WHERE url_token = ?;
+      `,
+      [challenge]
+    );
+    if (row) {
+      // row.start_date_utc is utc time (timezone offset is 0);
+      // startDateServer = start_date_utc + backend timezone offset (in minutes);
+      const startDateServer = new Date(row.start_date_utc);
+      const startDateUtc = new Date(
+        startDateServer.valueOf() -
+          startDateServer.getTimezoneOffset() * 60 * 1000
+      );
+      hasChallengeEnded =
+        new Date().valueOf() >
+        startDateUtc.valueOf() + 21 * 24 * 60 * 60 * 1000;
```

Comment:
I think this would be clearer (and more efficient) if we put the line:

```
const THREE_WEEKS = 3 * 7 * 24 * 60 * 60 * 1000;
```

at the top of the file, then used that constant here.